Add docs about skip ns propagation and PropagationPolicy/ClusterPropagationPolicy priority (#259)
karmada-io/website: docs/userguide/scheduling/resource-propagating.md

# emptymatch.yaml
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: propagation-emptymatch
spec:
  resourceSelectors:
    - apiVersion: apps/v1
      kind: Deployment
  placement:
    clusterAffinity:
      clusterNames:
        - member1
---
# labelselectormatch.yaml
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: propagation-labelselectormatch
spec:
  resourceSelectors:
    - apiVersion: apps/v1
      kind: Deployment
      labelSelector:
        matchLabels:
          app: nginx
  placement:
    clusterAffinity:
      clusterNames:
        - member2
---
# namematch.yaml
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: propagation-namematch
spec:
  resourceSelectors:
    - apiVersion: apps/v1
      kind: Deployment
      name: nginx
  placement:
    clusterAffinity:
      clusterNames:
        - member3
```
The `nginx` deployment in `default` namespace will be propagated to cluster `member3`.

### Select from same priority PropagationPolicy
In the case of PropagationPolicy matches, if the explicit priorities are the same, the one with the highest implicit priority will be selected.  

> Here is an example, if there are following policies:  

Reviewer: ```suggestion
Assume there are multiple policies:  
```

```yaml
# explicit-labelselectormatch.yaml
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: propagation-explicit-labelselectormatch
spec:
  resourceSelectors:
    - apiVersion: apps/v1
      kind: Deployment
      labelSelector:
        matchLabels:
          app: nginx
  priority: 3
  placement:
    clusterAffinity:
      clusterNames:
        - member1
---
# explicit-namematch.yaml
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: propagation-explicit-namematch
spec:
  resourceSelectors:
    - apiVersion: apps/v1
      kind: Deployment
      name: nginx
  priority: 3
  placement:
    clusterAffinity:
      clusterNames:
        - member3
```
The `nginx` deployment in `default` namespace will be propagated to cluster `member3`.

> If both explicit and implicit priorities are the same, Karmada applies the PropagationPolicy with lower alphabetical order, for example, choosing xxx-a-xxx instead of xxx-b-xxx.  

Reviewer: ```suggestion
If both explicit and implicit priorities are the same, Karmada applies the PropagationPolicy with lower alphabetical order, for example, choosing xxx-a-xxx instead of xxx-b-xxx.  
```
```suggestion
If both explicit and implicit priorities are the same, Karmada applies the PropagationPolicy in an ascending alphabetical order, for example, choosing xxx-a-xxx instead of xxx-b-xxx.  
```

> If the explicit/implicit priorities are all the same, select the one with the lower PropagationPolicy name alphabetical.  

Reviewer: ```suggestion
If both explicit and implicit priorities are the same, Karmada applies the PropagationPolicy whose first letter in name comes later in the alphabet.
```

> Here is an example, if there are following policies:  

Reviewer: ```suggestion
Assume there are multiple policies:  
```

    ...
```

> ## PropagationPolicy/ClusterPropagationPolicy priority

Reviewer 2: A little rough. For a user who does not read the code, may have a series of questions:
1. What is "implicit priority"? By default it is?
2. How do I config explicit priority?
3. Typical use scenario?
4. I suggest adding a specific example.

Author: I add specific example, please take a look

If multiple PropagationPolicy/ClusterPropagationPolicy matches the workload, Karmada will select the one with the highest priority. In PropagationPolicy/ClusterPropagationPolicy, implicit priority and explicit priority can be configured.

For explicit priority, the higher the number, the higher the priority.

For implicit priority, the priority order(from low to high) is as follows:
* The PropagationPolicy/ClusterPropagationPolicy resourceSelector whose name and labelSelector are empty matches the workload.

> * PropagationPolicy/ClusterPropagationPolicy labelSelector of resourceSelector matches the workload.

Reviewer 2: Grammar check.

Author: fix
